client/my-sites/domains/domain-search/index.jsx

```diff
@@ -89,13 +89,23 @@ class DomainSearch extends Component {
 	};
 
 	handleAddMapping = ( domain ) => {
-		addItem( domainMapping( { domain } ) );
-		page( '/checkout/' + this.props.selectedSiteSlug );
+		this.props.shoppingCartManager
+			.addProductsToCart( [
+				fillInSingleCartItemAttributes( domainMapping( { domain } ), this.props.productsList ),
+			] )
+			.then( () => {
+				page( '/checkout/' + this.props.selectedSiteSlug );
```

Comment:
Theoretically, these `then` handlers should guard if the component is already unmounted. If I navigate away from the Domains page while the cart update REST request is still running, the `then` handler will access `props` of an unmounted component (React will warn about that) and redirect me to checkout.

Reply:
Good point. I added a `isMounted` guard which hopefully will prevent this leak.